sh2lib/examples/http2_request/pytest_http2_request.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+#
+# SPDX-FileCopyrightText: 2021-2022 Espressif Systems (Shanghai) CO LTD
+# SPDX-License-Identifier: Apache-2.0
+
+import http.client
+import logging
+import os
+
+import pytest
+from pytest_embedded import Dut
+
+HTTP_OK = 200
+TEST_SERVER = 'http2.github.io'
+
+
+def is_test_server_available():  # type: () -> bool
+    # 443 - default https port
+    try:
+        conn = http.client.HTTPSConnection(TEST_SERVER, 443, timeout=10)
+        conn.request('GET', '/')
+        resp = conn.getresponse()
+        return True if resp.status == HTTP_OK else False
+    except Exception as msg:
+        logging.info('Exception occurred when connecting to {}: {}'.format(TEST_SERVER, msg))
+        return False
+    finally:
+        conn.close()
+
+
+@pytest.mark.esp32
+@pytest.mark.ethernet
+def test_examples_protocol_http2_request(dut: Dut) -> None:
```

Comment:
from idf_build_apps.constants import IDF_VERSION_MAJOR

@pytest.mark.skipif(IDF_VERSION_MAJOR < 4, reason='does not support old releases')

Reply:
Or we can use `--collect-app-info` with `idf-build-apps` while building the examples and then use `app-info-filepattern` while running the examples. This is the way we do in ESP-IDF CI?

Reply:
Also need to modify the `conftest.py`, which will make it more complicated. I don't recommend following the ESP-IDF CI in this repo, that's too complex.

Reply:
Thanks, added the change.